src/main/java/teammates/ui/webapi/FeedbackSessionLogsUpdateAction.java

```diff
@@ -0,0 +1,91 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import teammates.common.datatransfer.attributes.FeedbackSessionLogEntryAttributes;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Logger;
+
+/**
+ * Action: Sync feedback session logs from GCloud logging service.
+ */
+public class FeedbackSessionLogsUpdateAction extends AdminOnlyAction {
+
+    private static final Logger log = Logger.getLogger();
+    private static final int MIN_WINDOW_PERIOD = 2 * 1000;
+    private static final int MAX_WINDOW_SIZE = 1;
+
+    @Override
+    public ActionResult execute() {
+        Instant currentTime = Instant.now();
+        Map<String, Map<String, List<FeedbackSessionLogEntryAttributes>>> studentToLogs =
+                new HashMap<>();
+        List<FeedbackSessionLogEntryAttributes> validLogEntries = new ArrayList<>();
+        List<FeedbackSessionLogEntryAttributes> allLogEntries =
+                logsProcessor.getFeedbackSessionLogs(null, null,
+                        currentTime.minus(15, ChronoUnit.MINUTES).toEpochMilli(), currentTime.toEpochMilli(), null);
+
+        // Arrange logs based on student email and log type.
+        for (FeedbackSessionLogEntryAttributes logEntry : allLogEntries) {
+            String studentEmail = logEntry.getStudentEmail();
+            String logType = logEntry.getFeedbackSessionLogType();
+            Map<String, List<FeedbackSessionLogEntryAttributes>> studentLog =
+                    studentToLogs.getOrDefault(studentEmail, new HashMap<>());
+
+            List<FeedbackSessionLogEntryAttributes> currList =
+                    studentLog.getOrDefault(logType, new ArrayList<>());
+
+            currList.add(logEntry);
+            studentLog.put(logType, currList);
+            studentToLogs.put(studentEmail, studentLog);
+        }
+
+        for (Map.Entry<String, Map<String, List<FeedbackSessionLogEntryAttributes>>>
+                studentToLog : studentToLogs.entrySet()) {
+            Map<String, List<FeedbackSessionLogEntryAttributes>> studentLog =
+                    studentToLog.getValue();
+            int windowSize = 0;
+
+            for (Map.Entry<String, List<FeedbackSessionLogEntryAttributes>>
+                    logEntries : studentLog.entrySet()) {
+                List<FeedbackSessionLogEntryAttributes> logs = logEntries.getValue();
+                FeedbackSessionLogEntryAttributes startLog = logs.get(0);
+
+                for (FeedbackSessionLogEntryAttributes currLog : logs) {
+                    if (currLog.getTimestamp() - startLog.getTimestamp() < MIN_WINDOW_PERIOD) {
+                        windowSize++;
+                    } else {
+                        // Take the first log of the window.
+                        if (windowSize > MAX_WINDOW_SIZE) {
+                            startLog.setWindowSize(windowSize);
+                        }
```

Comment:
- I don't believe you need to define` MAX_WINDOW_SIZE` explicitly. The condition is simply "if there are multiple logs", not "if there is > x amount of logs, where x = 1".
- In fact, this check should not even be needed, similarly for down there. `startLog.setWindowSize(1)` is itself accurate.